specs/Deployment/DeploymentAPI.md

```diff
@@ -3,101 +3,117 @@
+application. The Deployment API enables a developer to check when the required WinAppSDK Main and
+Singleton packages are missing and get them installed.
 
-Since the Deployment API is part of the WinAppSDK framework package, the framework itself must be
-present to use the API. To deploy the WinAppSDK framework package with the MSIX packaged
-application, the framework must be declared as a dependency in the application manifest with a min
-version specified. For instructions on deploying the framework package, see the
-[developer docs](https://docs.microsoft.com/windows/apps/windows-app-sdk/deploy-packaged-apps#deploy-the-windows-app-sdk-framework-package).
+Since the Deployment API is part of the WinAppSDK Framework package, the Framework itself must be
+present to use the API. To deploy the WinAppSDK Framework package with the MSIX packaged
+application, the Framework must be declared as a dependency in the application manifest with a min
+version specified. For instructions on deploying the Framework package, see the
+[developer docs](https://docs.microsoft.com/windows/apps/windows-app-sdk/deploy-packaged-apps#deploy-the-windows-app-sdk-Framework-package).
 
-If a developer prefers to have the Microsoft Store automatically service the WinAppSDK framework
-package and if features like push notifications are desired, then the WinAppSDK main and singleton
-should also be deployed. Developers must not assume that the WinAppSDK main and singleton packages
-are already present on the system, or that the version present is the minimum version required by
-the application. Instead, the Deployment API should be used to validate that the WinAppSDK is at the
-correct version and that all required packages are present.
+If a developer prefers to have the Microsoft Store automatically service the WinAppSDK Framework
+package and if features like dynamic dependencies, push notifications are desired, then the
+WinAppSDK Main and Singleton should also be deployed. Developers must not assume that the WinAppSDK
+Main and Singleton packages are already present on the system, or that the version present is the
+minimum version required by the application. Instead, the Deployment API should be used to validate
+that the WinAppSDK is at the correct version and that all required packages are present.
 
-The Deployment API addresses the following two developer scenarios:
+The Deployment API addresses the following developer scenarios:
 
-1. Does this system have the min version of the WinAppSDK main and singleton packages required by my
+1. Does this system have the min version of the WinAppSDK Main and Singleton packages required by my
    app?
-2. How do I install the WinAppSDK main and singleton packages, if they are not already present on
+2. How do I install the WinAppSDK Main and Singleton packages, if they are not already present on
    the system?
+3. How do I repair the already installed WinAppSDK Main and Singleton packages, if needed ?
```

Comment:
Why do we need Repair to be a distinct action than Install/Initialize, instead of having developers [re]install the packages whenever they're "broken".  This is something of a breaking change since we'd previously told developers to Initialize for any status other than .OK, right?

The manual method to repair by "reinstalling from msdn downloads page" is identical for install and repair.  Why is the full-trust-app method different?

Reply:
Intialize installs/re-installs WinAppSDK Main and Singleton packages only when needed (i.e. when one of the packages is not installed or if one of them is in identified bad state). Thus, avoiding the unnecessary perf overhead of the unconditional re-install in every launch of the WinAppSDK dependent Packaged app.

Repair, on the other hand, re-installs those packages unconditionally. This API is important because there are many bad states a package can be in but only a few of them could be identified by Initialize and fix. Hence, we need Developers to just able to call Repair in cases where the WinAppSDK features aren't working due to a bad state that Initialize couldn't identify and fix it.